### PR TITLE
DPD France update

### PR DESCRIPTION
**Pull request by mbrentini**
MANDATORY : rename repo 'exapaq' to 'dpdfrance'

### CODE REVIEW

File: dpdfrance.php [hunk 1]
```diff
@@ -227,24 +227,24 @@ public function __construct()
 
 		parent::__construct();
 
-		$this->displayName = $this->l('EXAPAQ');
-		$this->description = $this->l('Offer EXAPAQ\'s fast and reliable delivery services to your customers');
+		$this->displayName = $this->l('DPD');
```
Comment (Quetzacoalt91) [inline dpdfrance.php]: We have multiple DPD modules on PrestaShop (DPD Poland, DPD Geopost, and now this one). Can you rename the module displayName to make it specific to the France ?

**Comment by Quetzacoalt91** [comment]
Hi @mbrentini,

You changed every configuration data for this new version. In order to keep the data during the module upgrade, it would be good to make an upgrade file which will convert every EXAPAQ_* to its DPDFRANCE_* equivalent.

If needed, here is some documentation: http://doc.prestashop.com/display/PS15/Auto-updating+modules

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

The module is renamed DPD France.

Is this upgrade file correct ?
```
<?php
/**
* 2007-2014 PrestaShop
*
* NOTICE OF LICENSE
*
* This source file is subject to the Academic Free License (AFL 3.0)
* that is bundled with this package in the file LICENSE.txt.
* It is also available through the world-wide-web at this URL:
* http://opensource.org/licenses/afl-3.0.php
* If you did not receive a copy of the license and are unable to
* obtain it through the world-wide-web, please send an email
* to license@prestashop.com so we can send you a copy immediately.
*
* DISCLAIMER
*
* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
* versions in the future. If you wish to customize PrestaShop for your
* needs please refer to http://www.prestashop.com for more information.
*
* @author    DPD S.A.S. <ensavoirplus.ecommerce@dpd.fr>
* @copyright 2015 DPD S.A.S.
* @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*/

if (!defined('_PS_VERSION_'))
	exit;

function upgrade_module_5_1_0($object)
{
	// Update DB Configuration values EXAPAQ_* to DPDFRANCE_*
	Configuration::updateValue('DPDFRANCE_NOM_EXP', Configuration::getValue('EXAPAQ_NOM_EXP'));
	Configuration::updateValue('DPDFRANCE_ADDRESS_EXP', Configuration::getValue('EXAPAQ_ADDRESS_EXP'));
	Configuration::updateValue('DPDFRANCE_ADDRESS2_EXP', Configuration::getValue('EXAPAQ_ADDRESS2_EXP'));
	Configuration::updateValue('DPDFRANCE_CP_EXP', Configuration::getValue('EXAPAQ_CP_EXP'));
	Configuration::updateValue('DPDFRANCE_VILLE_EXP', Configuration::getValue('EXAPAQ_VILLE_EXP'));
	Configuration::updateValue('DPDFRANCE_TEL_EXP', Configuration::getValue('EXAPAQ_TEL_EXP'));
	Configuration::updateValue('DPDFRANCE_EMAIL_EXP', Configuration::getValue('EXAPAQ_EMAIL_EXP'));
	Configuration::updateValue('DPDFRANCE_GSM_EXP', Configuration::getValue('EXAPAQ_GSM_EXP'));
	Configuration::updateValue('DPDFRANCE_RELAIS_DEPOT_CODE', Configuration::getValue('EXAPAQ_ICIRELAIS_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_RELAIS_SHIPPER_CODE', Configuration::getValue('EXAPAQ_ICIRELAIS_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_PREDICT_DEPOT_CODE', Configuration::getValue('EXAPAQ_PREDICT_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_PREDICT_SHIPPER_CODE', Configuration::getValue('EXAPAQ_PREDICT_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_CLASSIC_DEPOT_CODE', Configuration::getValue('EXAPAQ_CLASSIC_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_CLASSIC_SHIPPER_CODE', Configuration::getValue('EXAPAQ_CLASSIC_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_SUPP_ILES', Configuration::getValue('EXAPAQ_SUPP_ILES'));
	Configuration::updateValue('DPDFRANCE_SUPP_MONTAGNE', Configuration::getValue('EXAPAQ_SUPP_MONTAGNE'));
	Configuration::updateValue('DPDFRANCE_ETAPE_EXPEDITION', Configuration::getValue('EXAPAQ_ETAPE_EXPEDITION'));
	Configuration::updateValue('DPDFRANCE_ETAPE_EXPEDIEE', Configuration::getValue('EXAPAQ_ETAPE_EXPEDIEE'));
	Configuration::updateValue('DPDFRANCE_ETAPE_LIVRE', Configuration::getValue('EXAPAQ_ETAPE_LIVRE'));
	Configuration::updateValue('DPDFRANCE_AD_VALOREM', Configuration::getValue('EXAPAQ_AD_VALOREM'));
	Configuration::updateValue('DPDFRANCE_RELAIS_MYPUDO_URL', Configuration::getValue('EXAPAQ_ICIRELAIS_MYPUDO_URL'));
	return true;
}
?>
```

**Comment by Quetzacoalt91** [comment]
Hi @mbrentini,

Yes it is a good start. We could go farther by deleting the old keys with:
```php
Configuration::deleteByName('EXAPAQ_NOM_EXP');
[...]
```

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

If I understand well, upgrading a module doesn't reinstall it. As we need to add a new database table for the module, and its creation is done when install is triggered, should I add it in the upgrade script ?

Giving : 

```
<?php
/**
* 2007-2014 PrestaShop
*
* NOTICE OF LICENSE
*
* This source file is subject to the Academic Free License (AFL 3.0)
* that is bundled with this package in the file LICENSE.txt.
* It is also available through the world-wide-web at this URL:
* http://opensource.org/licenses/afl-3.0.php
* If you did not receive a copy of the license and are unable to
* obtain it through the world-wide-web, please send an email
* to license@prestashop.com so we can send you a copy immediately.
*
* DISCLAIMER
*
* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
* versions in the future. If you wish to customize PrestaShop for your
* needs please refer to http://www.prestashop.com for more information.
*
* @author    DPD S.A.S. <ensavoirplus.ecommerce@dpd.fr>
* @copyright 2015 DPD S.A.S.
* @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
*/

if (!defined('_PS_VERSION_'))
	exit;

function upgrade_module_5_1_0($object)
{
	// Update DB configuration values EXAPAQ_* to DPDFRANCE_*
	Configuration::updateValue('DPDFRANCE_NOM_EXP', Configuration::getValue('EXAPAQ_NOM_EXP'));
	Configuration::updateValue('DPDFRANCE_ADDRESS_EXP', Configuration::getValue('EXAPAQ_ADDRESS_EXP'));
	Configuration::updateValue('DPDFRANCE_ADDRESS2_EXP', Configuration::getValue('EXAPAQ_ADDRESS2_EXP'));
	Configuration::updateValue('DPDFRANCE_CP_EXP', Configuration::getValue('EXAPAQ_CP_EXP'));
	Configuration::updateValue('DPDFRANCE_VILLE_EXP', Configuration::getValue('EXAPAQ_VILLE_EXP'));
	Configuration::updateValue('DPDFRANCE_TEL_EXP', Configuration::getValue('EXAPAQ_TEL_EXP'));
	Configuration::updateValue('DPDFRANCE_EMAIL_EXP', Configuration::getValue('EXAPAQ_EMAIL_EXP'));
	Configuration::updateValue('DPDFRANCE_GSM_EXP', Configuration::getValue('EXAPAQ_GSM_EXP'));
	Configuration::updateValue('DPDFRANCE_RELAIS_DEPOT_CODE', Configuration::getValue('EXAPAQ_ICIRELAIS_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_RELAIS_SHIPPER_CODE', Configuration::getValue('EXAPAQ_ICIRELAIS_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_PREDICT_DEPOT_CODE', Configuration::getValue('EXAPAQ_PREDICT_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_PREDICT_SHIPPER_CODE', Configuration::getValue('EXAPAQ_PREDICT_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_CLASSIC_DEPOT_CODE', Configuration::getValue('EXAPAQ_CLASSIC_DEPOT_CODE'));
	Configuration::updateValue('DPDFRANCE_CLASSIC_SHIPPER_CODE', Configuration::getValue('EXAPAQ_CLASSIC_SHIPPER_CODE'));
	Configuration::updateValue('DPDFRANCE_SUPP_ILES', Configuration::getValue('EXAPAQ_SUPP_ILES'));
	Configuration::updateValue('DPDFRANCE_SUPP_MONTAGNE', Configuration::getValue('EXAPAQ_SUPP_MONTAGNE'));
	Configuration::updateValue('DPDFRANCE_ETAPE_EXPEDITION', Configuration::getValue('EXAPAQ_ETAPE_EXPEDITION'));
	Configuration::updateValue('DPDFRANCE_ETAPE_EXPEDIEE', Configuration::getValue('EXAPAQ_ETAPE_EXPEDIEE'));
	Configuration::updateValue('DPDFRANCE_ETAPE_LIVRE', Configuration::getValue('EXAPAQ_ETAPE_LIVRE'));
	Configuration::updateValue('DPDFRANCE_AD_VALOREM', Configuration::getValue('EXAPAQ_AD_VALOREM'));
	Configuration::updateValue('DPDFRANCE_RELAIS_MYPUDO_URL', Configuration::getValue('EXAPAQ_ICIRELAIS_MYPUDO_URL'));

	// Delete old DB configuration values
	Configuration::deleteByName('EXAPAQ_NOM_EXP');
	Configuration::deleteByName('EXAPAQ_ADDRESS_EXP');
	Configuration::deleteByName('EXAPAQ_ADDRESS2_EXP');
	Configuration::deleteByName('EXAPAQ_CP_EXP');
	Configuration::deleteByName('EXAPAQ_VILLE_EXP');
	Configuration::deleteByName('EXAPAQ_TEL_EXP');
	Configuration::deleteByName('EXAPAQ_EMAIL_EXP');
	Configuration::deleteByName('EXAPAQ_GSM_EXP');
	Configuration::deleteByName('EXAPAQ_ICIRELAIS_CARRIER_ID', '');
	Configuration::deleteByName('EXAPAQ_ICIRELAIS_DEPOT_CODE', '');
	Configuration::deleteByName('EXAPAQ_ICIRELAIS_SHIPPER_CODE', '');
	Configuration::deleteByName('EXAPAQ_PREDICT_CARRIER_ID', '');
	Configuration::deleteByName('EXAPAQ_PREDICT_DEPOT_CODE', '');
	Configuration::deleteByName('EXAPAQ_PREDICT_SHIPPER_CODE', '');
	Configuration::deleteByName('EXAPAQ_CLASSIC_CARRIER_ID', '');
	Configuration::deleteByName('EXAPAQ_CLASSIC_DEPOT_CODE', '');
	Configuration::deleteByName('EXAPAQ_CLASSIC_SHIPPER_CODE', '');
	Configuration::deleteByName('EXAPAQ_ICIRELAIS_MYPUDO_URL');
	Configuration::deleteByName('EXAPAQ_SUPP_ILES');
	Configuration::deleteByName('EXAPAQ_SUPP_MONTAGNE');
	Configuration::deleteByName('EXAPAQ_ETAPE_EXPEDITION');
	Configuration::deleteByName('EXAPAQ_ETAPE_EXPEDIEE');
	Configuration::deleteByName('EXAPAQ_ETAPE_LIVRE');
	Configuration::deleteByName('EXAPAQ_AD_VALOREM');
	Configuration::deleteByName('EXAPAQ_PARAM'))

	// Install new DB table
	$query = Db::getInstance()->Execute('CREATE TABLE IF NOT EXISTS `'._DB_PREFIX_.'dpdfrance_shipping` (
		`id_customer` int(10) unsigned DEFAULT NULL,
		`id_cart` int(10) unsigned DEFAULT NULL,
		`id_carrier` int(5) unsigned DEFAULT NULL,
		`service` varchar(3) DEFAULT NULL,
		`relay_id` varchar(8) DEFAULT NULL,
		`company` varchar(23) DEFAULT NULL,
		`address1` varchar(128) DEFAULT NULL,
		`address2` varchar(128) DEFAULT NULL,
		`postcode` varchar(10) DEFAULT NULL,
		`city` varchar(100) DEFAULT NULL,
		`id_country` int(11) DEFAULT NULL,
		`gsm_dest` varchar(14) DEFAULT NULL) 
		ENGINE=MyISAM DEFAULT CHARSET=utf8;');

	if (!$query)
		return false;

	return true;
}
?>
```

**Comment by Quetzacoalt91** [comment]
@mbrentini,

Yes exactly, a upgrade is totally different from an uninstall/reinstall. You have to manage the upgrade with a specific file.
In consequence, you did well. You have to add your CREATE TABLE here :)

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

I just pushed a new commit : aca47a1 which includes the upgrade file.
How could we test it ?

Thanks

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

I added a drop table line to get rid of the exapaq_france db table.

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

One question : how do we deal with the folder changing its name from 'exapaq' to 'dpdfrance' ?

**Comment by Quetzacoalt91** [comment]
Actually, this gonna be difficult. I just discovered that building a upgrade file is useless if we change the module name ... because the merchant WILL HAVE to install a new module.
But, don't worry, creating the upgrade was not a waste of time. We can move a good part of its content in the install function to migrate the configuration when the new one will be installed. :)

**Comment by mbrentini** [comment]
Well, I think the best would be to create a new repo named 'dpdfrance' and push the new module there. Could you do that ? 

Thank you

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

This commit 771fa2b should be correct for customer upgrading from EXAPAQ to DPD France.
It copies the DB configuration values, deletes old carriers and exapaq specific tables, and disables the EXAPAQ module.

The customer just has to recreate his carriers (mandatory b/c of logos, tracking URL changing) and his shipping rates.

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

This commit 5f8bd6e aims to ensure backwards compatibility with PS 1.3 and 1.4, as some classes are missing.

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

This commit 05eec02 also uninstalls old EXAPAQ and ICI relais modules, saving configuration for customers coming from EXAPAQ v5 modules.

**Comment by Quetzacoalt91** [comment]
Hi @mbrentini,

Thanks, your PR has just been merged and the "new" module should be available from tomorrow.

Regards

**Pull request merged**